src/main/java/org/zalando/fahrschein/PersistentCursorManager.java

```diff
@@ -8,36 +8,51 @@
 import javax.sql.DataSource;
 import java.io.IOException;
 import java.util.Collection;
+import java.util.Optional;
+
+import static java.text.MessageFormat.format;
 
 public class PersistentCursorManager implements CursorManager {
 
     private static final Logger LOG = LoggerFactory.getLogger(PersistentCursorManager.class);
 
     private final JdbcTemplate template;
     private final String consumerName;
+    private final Optional<String> schema;
 
-    public PersistentCursorManager(JdbcTemplate template, String consumerName) {
+    public PersistentCursorManager(final JdbcTemplate template, final String schema, final String consumerName) {
         this.template = template;
+        this.schema = Optional.ofNullable(schema);
         this.consumerName = consumerName;
     }
 
-    public PersistentCursorManager(DataSource dataSource, String consumerName) {
+    public PersistentCursorManager(final JdbcTemplate template, final String consumerName) {
+        this.template = template;
+        this.consumerName = consumerName;
+        this.schema = Optional.empty();
+    }
+
+    public PersistentCursorManager(final DataSource dataSource, final String consumerName) {
         this(new JdbcTemplate(dataSource), consumerName);
     }
 
     @Override
-    public void onSuccess(String eventName, Cursor cursor) throws IOException {
-        template.queryForObject("SELECT * FROM nakadi_cursor_update(?, ?, ?, ?)", new Object[]{consumerName, eventName, cursor.getPartition(), cursor.getOffset()}, Integer.class);
+    public void onSuccess(final String eventName, final Cursor cursor) throws IOException {
+        final String sql = format("SELECT {0}nakadi_cursor_update(?, ?, ?, ?)", schema.map(s -> s + ".").orElse(""));
```

Comment:
> Do you find this more readable?

Not because of readability, but because message format is mostly used for i18n or l10n, the java doc even says

> MessageFormat provides a means to produce concatenated messages in a language-neutral way. **Use this to construct messages displayed for end users.**